x-pack/packages/kbn-elastic-assistant/impl/assistant/quick_prompts/quick_prompts.tsx

```diff
@@ -90,49 +93,84 @@ export const QuickPrompts: React.FC<QuickPromptsProps> = React.memo(
     }, [setIsSettingsModalVisible, setSelectedSettingsTab]);
 
     return (
-      <QuickPromptsFlexGroup gutterSize="s" alignItems="center">
-        {contextFilteredQuickPrompts.slice(0, COUNT_BEFORE_OVERFLOW).map((badge, index) => (
-          <EuiFlexItem key={index} grow={false}>
-            <EuiBadge
-              color={badge.color}
-              onClick={() => onClickAddQuickPrompt(badge)}
-              onClickAriaLabel={badge.title}
-            >
-              {badge.title}
-            </EuiBadge>
-          </EuiFlexItem>
-        ))}
-        {contextFilteredQuickPrompts.length > COUNT_BEFORE_OVERFLOW && (
-          <EuiFlexItem grow={false}>
-            <EuiPopover
-              button={
+      <EuiFlexGroup
+        gutterSize="s"
+        alignItems="center"
+        justifyContent={isFlyoutMode ? 'spaceBetween' : 'flexStart'}
+        css={
+          !isFlyoutMode &&
+          css`
+            margin: 16px;
+          `
+        }
+      >
+        <EuiFlexItem
+          grow={false}
+          css={css`
+            overflow: hidden;
+          `}
+        >
+          <EuiFlexGroup gutterSize="s" alignItems="center" wrap={false}>
+            {contextFilteredQuickPrompts.slice(0, COUNT_BEFORE_OVERFLOW).map((badge, index) => (
```

Comment:
Probably need to make the quick prompt overflow be width dependent instead of count of badges now:

<img width="449" alt="image" src="https://github.com/elastic/kibana/assets/2946766/799c7e51-799d-49ea-b3b3-97cc7c402cd6">